vscode/src/debugger/session.ts

```diff
@@ -144,57 +190,392 @@ export class QscDebugSession extends LoggingDebugSession {
+      breakpoints: [],
+    };
+
+    const fileUri = vscode.Uri.parse(args.source.path ?? "", false);
+
+    const file = vscode.workspace.textDocuments.find(
+      (td) => td.uri.path === fileUri.path
+    );
+    if (!file) {
+      for (const td of vscode.workspace.textDocuments) {
+        log.trace("breakpointLocationsRequest: potential file" + td.uri.path);
+      }
+      log.trace("breakpointLocationsRequest: target file" + fileUri.path);
+    }
+    if (fileUri && file) {
+      // Map request start/end line/column to file offset for debugger
+      const lineRange = file.lineAt(args.line).range;
+      const startLine = lineRange.start.line;
+      const startCol = args.column
+        ? this.convertClientColumnToDebugger(args.column)
+        : lineRange.start.character;
+      const endLine = args.endLine
+        ? this.convertClientLineToDebugger(args.endLine)
+        : lineRange.end.line;
+      const endCol = args.endColumn
+        ? this.convertClientColumnToDebugger(args.endColumn)
+        : lineRange.end.character;
+      const startPos = new vscode.Position(startLine, startCol);
+      const endPos = new vscode.Position(endLine, endCol);
+      const startOffset = file.offsetAt(startPos);
+      const endOffset = file.offsetAt(endPos);
+
+      log.trace(
+        `breakpointLocationsRequest: ${startLine}:${startCol} - ${endLine}:${endCol}`
+      );
+      log.trace(`breakpointLocationsRequest: ${startOffset} - ${endOffset}`);
+      // Now that we have the mapped breakpoint span, get the actual breakpoints
+      // from the debugger
+      // This currently has issues with breakpoints that span multiple lines
+      // stmt for example may have a span that only includes the identifier
+      // where the rest of the statement is on the next line(s)
+      /*const bps =
+        this._breakpointLocations
+          .get(fileUri.path)
+          ?.filter((bp) => startOffset <= bp.lo && bp.hi <= endOffset) ?? [];*/
+      const bps = this.breakpointLocations.get(fileUri.path) ?? [];
+      log.trace(`breakpointLocationsRequest: candidates %O`, bps);
+
+      // must map the debugger breakpoints back to the client breakpoint locations
+      const bls = bps.map((bps) => {
+        const startPos = file.positionAt(bps.lo);
+        const endPos = file.positionAt(bps.hi);
+        const bp: DebugProtocol.BreakpointLocation = {
+          line: this.convertDebuggerLineToClient(startPos.line),
+          column: this.convertDebuggerColumnToClient(startPos.character),
+          endLine: this.convertDebuggerLineToClient(endPos.line),
+          endColumn: this.convertDebuggerColumnToClient(endPos.character),
+        };
+        return bp;
+      });
+      log.trace(`breakpointLocationsRequest: mapped %O`, bls);
+      response.body = {
+        breakpoints: bls,
+      };
+    }
+    log.trace(`breakpointLocationsResponse: %O`, response);
+    this.sendResponse(response);
+  }
+
+  protected async setBreakPointsRequest(
```

Comment:
Minor nit, but as you can see from the DebugProtocol types, 'Breakpoint' is one word, not 'BreakPoint'.

Reply:
This is an override for a base method. I didn't choose the name.